public/app/features/explore/QueryField.tsx

```diff
@@ -295,14 +304,28 @@ class QueryField extends React.PureComponent<TypeaheadFieldProps, TypeaheadField
 
   applyTypeahead(change: Change, suggestion: Suggestion): Change {
     const { cleanText, onWillApplySuggestion, syntax } = this.props;
-    const { typeaheadPrefix, typeaheadText } = this.state;
+    const { typeaheadPrefix, typeaheadText, placeholderIndices } = this.state;
     let suggestionText = suggestion.insertText || suggestion.label;
-    const move = suggestion.move || 0;
+    let move = suggestion.move || 0;
 
     if (onWillApplySuggestion) {
       suggestionText = onWillApplySuggestion(suggestionText, { ...this.state });
     }
 
+    const placeholderInsertIndex = placeholderIndices[0];
```

Comment:
I think this logic should in its own function, and not in `applyTypeahead` which is reserved for accepting a suggestion. Can you make a special case in the `Tab` key handler?